src/linear_algebra/affine_space.lean

```diff
@@ -1180,6 +1194,84 @@ end
 
 end affine_space
 
+namespace affine_subspace
+
+variables {k : Type*} {V : Type*} {P : Type*} [ring k] [add_comm_group V] [module k V]
+          [affine_space k V P]
+
+open affine_space
+
+/-- The direction of the sup of two nonempty affine subspaces is the
+sup of the two directions and of any one difference between points in
+the two subspaces. -/
+lemma direction_sup {s1 s2 : affine_subspace k V P} {p1 p2 : P} (hp1 : p1 ∈ s1) (hp2 : p2 ∈ s2) :
+  (s1 ⊔ s2).direction = s1.direction ⊔ s2.direction ⊔ submodule.span k {p2 -ᵥ p1} :=
+begin
+  refine le_antisymm _ _,
+  { change (affine_span k V ((s1 : set P) ∪ s2)).direction ≤ _,
+    rw ←mem_coe at hp1,
+    rw [direction_affine_span, vector_span_eq_span_vsub_set_right k V (set.mem_union_left _ hp1),
+        submodule.span_le],
+    rintros v ⟨p3, hp3, rfl⟩,
+    cases hp3,
+    { rw [sup_assoc, sup_comm, submodule.mem_coe, submodule.mem_sup],
+      use [0, submodule.zero_mem _, p3 -ᵥ p1, vsub_mem_direction hp3 hp1],
+      rw zero_add },
+    { rw [sup_assoc, submodule.mem_coe, submodule.mem_sup],
+      use [0, submodule.zero_mem _, p3 -ᵥ p1],
+      rw [and_comm, zero_add],
+      use rfl,
+      rw [←vsub_add_vsub_cancel V p3 p2 p1, submodule.mem_sup],
+      use [p3 -ᵥ p2, vsub_mem_direction hp3 hp2, p2 -ᵥ p1,
+           submodule.mem_span_singleton_self _] } },
+  { refine sup_le (sup_direction_le _ _) _,
+    rw [direction_eq_vector_span, vector_span_def],
+    exact Inf_le_Inf (λ p hp, set.subset.trans
+      (set.singleton_subset_iff.2
+        (vsub_mem_vsub_set V (mem_span_points k V p2 _ (set.mem_union_right _ hp2))
+                             (mem_span_points k V p1 _ (set.mem_union_left _ hp1))))
+      hp) }
+end
+
+/-- The direction of the span of the result of adding a point to a
+nonempty affine subspace is the sup of the direction of that subspace
+and of any one difference between that point and a point in the
+subspace. -/
+lemma direction_affine_span_insert {s : affine_subspace k V P} {p1 p2 : P} (hp1 : p1 ∈ s) :
+  (affine_span k V (insert p2 (s : set P))).direction = s.direction ⊔ submodule.span k {p2 -ᵥ p1} :=
```

Comment:
Should the rhs be swapped? It is `insert p2 s`, so on the rhs I would expect `p2` to the left of `s`. That will probably save some `sup_comm` down the road.

Reply:
Is the suggestion to swap the two arguments of the `⊔`? I don't currently have any `sup_comm` in code using this, but can swap the arguments if that's considered to make a cleaner statement.

Reply:
Yup, that's what I meant. I think `mem_insert` (for example) also puts `p2` on the left and `s` on the right. Of course it's a mathematical non-issue. But if we follow this idea in the whole library, things will fit better together in general.

Reply:
I did the swap. It ended up adding one `sup_comm` to the proof of this lemma and removing one `add_comm` from the proof of the following lemma after avoiding adding `sup_comm` there.

Reply:
Nice. I hope it will save another 10 `add_comm`s and `sup_comm`s in the future :smile: